### PR TITLE
Stabilize and re-export core::array in std

### LINKED ISSUE #60014: Stabilize the core::array module and reexport in std (for TryFromSliceError)
[`TryFromSliceError`](https://doc.rust-lang.org/nightly/core/array/struct.TryFromSliceError.html) has been stabilized in [`core::array`](https://doc.rust-lang.org/nightly/core/array/index.html) but `core::array` is an unstable module. Additionally `TryFromSliceError` isn't re-exported anywhere in `std`. There isn't even a `std::array` module to re-export it into.

cc @rust-lang/libs

### PR DESCRIPTION
**Pull request by JohnTitor**
Fixes #60014

### CODE REVIEW

File: src/libcore/array.rs [hunk 1]
```diff
@@ -30,13 +27,17 @@ use crate::slice::{Iter, IterMut};
 /// Note that the traits AsRef and AsMut provide similar methods for types that
 /// may not be fixed-size arrays. Implementors should prefer those traits
 /// instead.
+#[stable(feature = "fixed_size_array", since = "1.36.0")]
```
Comment (Centril) [inline src/libcore/array.rs]: I don't think T-libs signed off on stabilizing `FixedSizeArray`?

Comment (JohnTitor) [reply src/libcore/array.rs]: Oops, I misread?

Comment (Centril) [reply src/libcore/array.rs]: I believe they wanted to stabilize just the module and `TryFromSliceError`.

Comment (JohnTitor) [reply src/libcore/array.rs]: I got it, we should make them unstable?

Comment (Centril) [reply src/libcore/array.rs]: I believe so, yes.

Comment (JohnTitor) [reply src/libcore/array.rs]: Okay, thank you!

**Comment by SimonSapin** [comment]
Looks good, thanks!

@bors r+

**Comment by jethrogb** [comment]
You're missing some changes in the `sgx` platform, see https://gist.github.com/fortanix-bot/a355e5bad7852c6b19e58ddff11418f8

**Comment by JohnTitor** [comment]
@jethrogb Thank you for pointing out, what I should do is to add `#![feature(fixed_size_array)]` to the top in `src/libstd/sys/sgx/rwlock.rs`?

**Comment by jethrogb** [comment]
Well, features need to be added at the root of the crate, which you could do conditionally depending on the target (there's already a list for SGX). However, I haven't looked at this PR in detail, but since you're stabilizing, I'd image it's just a matter of changing to code to use the stable API instead?

**Comment by JohnTitor** [comment]
Oh, I missed it, `src/libstd/sys/sgx/mod.rs` instead? And this PR stabilizes `core::array`, not `fixed_size_array` so we should add feature attribute I think.

**Comment by jethrogb** [comment]
> Oh, I missed it, src/libstd/sys/sgx/mod.rs instead? 

No `src/libstd/libc.rs`.

**Comment by jethrogb** [comment]
I think you might get a warning about an unused feature now (but I'm not sure).

**Comment by jethrogb** [comment]
I think you might get a warning about an unused feature now on non-SGX platforms (but I'm not sure).

**Comment by jethrogb** [comment]
I think your PR might be merged at 028e78d because the failure only happens when compiling with `cfg(test, target_env = "sgx")` and Rust CI is not gated on that build. A new PR would then be needed to fix this.

**Comment by jethrogb** [comment]
So it turns out if you push to your branch while bors is testing it in a rollup is a bad idea ;) I think this PR can be closed. I've filed #60684 for the SGX failure

**Comment by JohnTitor** [comment]
Oops, sorry... I'll close this.

**Pull request closed without merging**